runtime/stdlib/random.go

```diff
@@ -34,37 +35,164 @@ NOTE: The use of this function is unsafe if not used correctly.
-					errors.WrapPanic(func() {
-						err = generator.ReadRandom(buffer[:])
-					})
-					if err != nil {
-						panic(interpreter.WrappedExternalError(err))
-					}
-					return binary.LittleEndian.Uint64(buffer[:])
-				},
-			)
+			inter := invocation.Interpreter
+
+			// TODO: Check if invocation has an argument and implement modulo operation.
+
+			returnIntegerType := invocation.TypeParameterTypes.Oldest().Value
+			returnIntegerStaticType := interpreter.ConvertSemaToStaticType(inter, returnIntegerType)
+
+			switch returnIntegerStaticType {
+			// UInt*
+			case interpreter.PrimitiveStaticTypeUInt8:
+				return interpreter.NewUInt8Value(
+					inter,
+					func() uint8 {
+						return getRandomBytes(generator, 1)[0]
+					},
+				)
+			case interpreter.PrimitiveStaticTypeUInt16:
+				return interpreter.NewUInt16Value(
+					inter,
+					func() uint16 {
+						return binary.LittleEndian.Uint16(getRandomBytes(generator, 2))
+					},
+				)
+			case interpreter.PrimitiveStaticTypeUInt32:
+				return interpreter.NewUInt32Value(
+					inter,
+					func() uint32 {
+						return binary.LittleEndian.Uint32(getRandomBytes(generator, 4))
+					},
+				)
+			case interpreter.PrimitiveStaticTypeUInt64:
+				return interpreter.NewUInt64Value(
+					inter,
+					func() uint64 {
+						return binary.LittleEndian.Uint64(getRandomBytes(generator, 8))
+					},
+				)
+			case interpreter.PrimitiveStaticTypeUInt128:
+				return interpreter.NewUInt128ValueFromBigInt(
+					inter,
+					func() *big.Int {
+						buffer := getRandomBytes(generator, 16)
+						return interpreter.LittleEndianBytesToUnsignedBigInt(buffer)
+					},
+				)
+			case interpreter.PrimitiveStaticTypeUInt256:
+				return interpreter.NewUInt256ValueFromBigInt(
+					inter,
+					func() *big.Int {
+						buffer := getRandomBytes(generator, 32)
+						return interpreter.LittleEndianBytesToUnsignedBigInt(buffer)
+					},
+				)
+
+			// Word*
+			case interpreter.PrimitiveStaticTypeWord8:
+				return interpreter.NewWord8Value(
+					inter,
+					func() uint8 {
+						return getRandomBytes(generator, 1)[0]
+					},
+				)
+			case interpreter.PrimitiveStaticTypeWord16:
+				return interpreter.NewWord16Value(
+					inter,
+					func() uint16 {
+						return binary.LittleEndian.Uint16(getRandomBytes(generator, 2))
+					},
+				)
+			case interpreter.PrimitiveStaticTypeWord32:
+				return interpreter.NewWord32Value(
+					inter,
+					func() uint32 {
+						return binary.LittleEndian.Uint32(getRandomBytes(generator, 4))
+					},
+				)
+			case interpreter.PrimitiveStaticTypeWord64:
+				return interpreter.NewWord64Value(
+					inter,
+					func() uint64 {
+						return binary.LittleEndian.Uint64(getRandomBytes(generator, 8))
+					},
+				)
+			case interpreter.PrimitiveStaticTypeWord128:
+				return interpreter.NewWord128ValueFromBigInt(
+					inter,
+					func() *big.Int {
+						buffer := getRandomBytes(generator, 16)
+						return interpreter.LittleEndianBytesToUnsignedBigInt(buffer)
```

Comment:
Are the bytes returned from `getRandomBytes`/ `generator.ReadRandom()` always little-endian (or is it a safe assumption)?
The API doesn't specify this. cc: @tarakby

Reply:
`ReadRandom` only generates bytes, it doesn't generate numbers. This means that there is not endianness notion at the level of `ReadRandom` and its output (that's why the doc doesn't talk about it).

Endianness only makes sense when we want to convert the bytes into a number (at `revertibleRandom` implementation level, since it currently returns a 64-bits number). For that purpose, I [chose little endianness](https://github.com/onflow/cadence/blob/25e5405ecf9a1a2c580155012975b1764281fc24/runtime/stdlib/random.go#L65) but we could also choose big. That choice changes the output numbers, but does not impact the random number distribution.

Reply:
oh I see. Thanks for the explanation!

@darkdrag00nv2 Maybe we could avoid the conversion here (in `LittleEndianBytesToUnsignedBigInt`) and directly use `BigEndianBytesToUnsignedBigInt`?

Reply:
Might also make sense to consider big-endian for all the byte-to-number conversion in that case, for consistency?

Reply:
Agreed! The `binary.(Little|Big)Endian.UintN` functions are useful to convert the bytes into an integer, but it doesn't matter which we use. Maybe consistently use big, and remove `LittleEndianBytesToUnsignedBigInt`

Reply:
Done